111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,232 @@
+Kaniko executor did everything at once, which can be done by Buildah as well, but I rejected to have it done in one command because:
+
+- We will not get the SHA of image after the push.
+- Once pushed, the image will not be stored locally, so we cannot check the SHA using some different command.
+
+The SHA of the image is needed to keep the compatibility with the previous implementation using Kaniko.
+Kaniko, once the image is build and pushed, is able to return the full name of the image together with the SHA to particular path.
+That is done using the `--image-name-with-digest-file` option of the Kaniko executor.
+Unfortunately, Buildah doesn't have such option in case that you want to build and push the image in one command (that is done using the `buildah build` command, in order to push directly, you need to prefix the image with `docker://`).
+Buildah has option to store the SHA to specify file on path using `--digestfile` option - but only when `push` command is used.
+It will just output the SHA, not the full image, so we need to build it from the SHA and built image and output it to `/dev/termination-log` in order to have it inside the message of the completed Pod.
+
+We will use following commands:
+
+```shell
+buildah build --file=/dockerfile/Dockerfile --tag=<IMAGE> --storage-driver=vfs
+buildah push --storage-driver=vfs --digestfile=/tmp/digest <ADDITIONAL_OPTS> <IMAGE>
+buildah images --digests --filter=digest=sha256:$(cat /tmp/digest) --format='{{.Name}}@{{.Digest}}' > /dev/termination-log
+```
+- `<IMAGE>` is placeholder for user desired name of the image (with registry, repository, and possibly tag)
+- `<ADDITIONAL_OPTS>` is placeholder for user desired additional options for the `buildah push` command
+
+We need to take these three steps in order to get the SHA of the image together with correct repository.
+This will work in all scenarios, including when the user does not specify a tag and the default (`latest`) is used.
+
+Together with this feature, and upcoming v1 API, we will do following changes inside the `DockerOutput` model:
+
+1. we will deprecate `additionalKanikoOptions` field as the name of it is connected directly to Kaniko and more generic name can be used (this field will be removed as part of the v1 API).
+2. we will add `additionalOptions` field which will be used from now on and will cover additional options for both Kaniko and Buildah.
+
+The `additionalKanikoOptions` will still be used until the Buildah feature gate is moved to Beta stage, but warning about deprecation will be added inside the `.status` section of the `KafkaConnect` CR.
+Additionally, if both of the fields will be filled with values, `additionalOptions` takes precedence.
+Once the `UseConnectBuildWithBuildah` feature gate is moved to Beta, the `additionalKanikoOptions` will be completely ignored, the warning about deprecation will slightly change to "is deprecated and ignored" in case that it will be used.
```

Comment:
I do not think this makes sense. Although I guess the `v1` will actually drop it around the same time if my proposal is approved.